### PR TITLE
Developer Mode (With Makefile support)

### PR DESCRIPTION
**Pull request by ahmadia**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Makefile for Domain Discovery Tool development
```
Comment (ahmadia) [inline Makefile:1]: This file should aid development environment setup and will replace the fabfile.

File: bin/ddt-dev [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
Comment (ahmadia) [inline bin/ddt-dev:1]: This script automatically sets up the environment and replaces run_demo.sh

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: ddt
```
Comment (ahmadia) [inline environment.yml:1]: This replaces the virtualenv setup and also manages the elasticsearch installation for you.

File: logs/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Logfiles from supervisor processes go here
```
Comment (ahmadia) [inline logs/README.md:1]: Support directory for logging of output by supervisor

File: supervisord.conf [hunk 5]
```diff
@@ -0,0 +1,28 @@
+[supervisorctl]
```
Comment (ahmadia) [inline supervisord.conf:1]: Supervisor configuration, it can also run (and log) DDT in development mode

**Comment by ahmadia** [comment]
@yamsgithub @julianafreire @kriehl @brittainhard @aecio - This is ready for review and testing.  I've added a note for each of the files I added, and I've done some documentation on the development environment here: https://github.com/ViDA-NYU/domain_discovery_tool/wiki/Developer-Notes

Brittain has helped me test this on OS X, and I did some testing on Linux, but I expect there may be a few bugs to shake out.  Let me know if you have any questions or encounter any issues.

**Comment by yamsgithub** [comment]
@julianafreire @kriehl @brittainhard @aecio - I tested the 'make' file version of the development environment on my linux machine and everything worked fine. I had to install miniconda which was easy. Should we go ahead with the merge?

**Comment by ahmadia** [comment]
Any concerns or comments?  Where should we keep/maintain documentation on building/installing DDT for local development?

**Comment by yamsgithub** [comment]
Actually I realized I did not check creating the docker package. Let me do that.

**Comment by ahmadia** [comment]
Hmn, I didn't touch the Dockerfile at all.  Since I didn't touch the fabfile, that should all still work.

**Comment by yamsgithub** [comment]
Tested the docker and it looks fine. I am however not familiar with supervisord. Will looks into it later. Otherwise I think all look good.

**Comment by yamsgithub** [comment]
Maybe we can maintain the development-notes in the same directory as Readme.md and have the readme point to the development notes.

**Comment by ahmadia** [comment]
I think that's a good idea.  I'll wait on the thumbs up from @kriehl or @julianafreire before merging this in.

**Comment by aecio** [comment]
Hey @ahmadia, I also tested here and it's working fine. I just added the path for the symbolic link into the `.gitignore`, as it doesn't need to be commited. I think this is ready to be merged.

There's any particular reason for you to set up elasticsearch to start automatically, but not to do the same for DDT in the supervisord config file?

I started a port of the Dockerfile to use conda instead  of the fabfile (https://github.com/ViDA-NYU/domain_discovery_tool/commit/a8905e9a900530c1ed04560f04d2202bf93aa207). It would be nice to use supervisord to manage the services, but sometimes we use an external elasticsearch instance (configuring the enviroment variable ELASTICSEARCH_SERVER), and sometimes we start an ElasticSearch instance inside the container (for the demos to be self-contained, for example). What do you think is the best way to set up this? Maybe to have two supervisor config files (one for each case)?

**Comment by ahmadia** [comment]
> There's any particular reason for you to set up elasticsearch to start automatically, but not to do the same for DDT in the supervisord config file?

No.  On the Memex Explorer side, Brittain likes to have explicit control of his Django test server, so he usually launches it in a separate terminal.  If you prefer to have everything running by default under supervisor, that's an easy fix.

> I started a port of the Dockerfile to use conda instead of the fabfile (a8905e9). It would be nice to use supervisord to manage the services, but sometimes we use an external elasticsearch instance (configuring the enviroment variable ELASTICSEARCH_SERVER), and sometimes we start an ElasticSearch instance inside the container (for the demos to be self-contained, for example). What do you think is the best way to set up this? Maybe to have two supervisor config files (one for each case)?

I don't have a strong opinion on this, but two supervisor configurations seems reasonable.  The supervisor configurations here are so simple that you could just copy/paste, but if you like, you could create a third "base" configuration and have both configurations [include](http://supervisord.org/configuration.html#include-section-settings) it.

**Comment by ahmadia** [comment]
@yamsgithub @aecio - Katrina and I are comfortable with you merging this in if it's working for both of you and you're using it.

**Comment by ahmadia** [comment]
Actually, I was wrong, we do start the server by default: https://github.com/memex-explorer/memex-explorer/blob/master/source/supervisord.conf#L56

@aecio - should I turn autostart back on before we merge this PR?

**Comment by aecio** [comment]
@ahmadia Yes, if you guys are ok with that, you can go ahead and turn it on, and merge. As DDT has only two services (cherrypy and ES), its easy to start them manually if one want to do so.

**Comment by ahmadia** [comment]
Done.

**Pull request merged**